Realm/RLMRealm.h

```diff
@@ -495,6 +496,22 @@ __deprecated_msg("Use `performMigrationForConfiguration:error:`") NS_REFINED_FOR
  */
 + (BOOL)performMigrationForConfiguration:(RLMRealmConfiguration *)configuration error:(NSError **)error;
 
+/**
+ Creates a `RLMThreadHandover` from the passed in objects that can be safely imported on a separate thread.
+
+ Used to pass objects between threads in a thread-safe manner.
+
+ @warning Note that the returned `ThreadHandover` must be imported at most once, and that the current version of the
+ Realm will remain pinned until this package is imported or deallocated.
+
+ @param objects The objects to pass between threads.
+ @return A package that can be imported on a separate thread.
+
+ @see RLMThreadHandover
+ */
+- (RLMThreadHandover *)exportThreadHandoverWithObjects:(NSArray<id<RLMThreadConfined>> *)objects
```

Comment:
I discussed the API over DM with @bdash. Here are some takeaways:

1) There is no need for the `ThreadHandover` to contain a `Realm.Configuration` and no reason for the import function to produce a `Realm`. The API is mostly designed as such for historical reasons (`realm.asyncWrite { realm, objects in … }`) where the `Realm` had to be created on the new thread. Given that we're making the API "less convenient", it might make sense to no longer pass the configuration. Anyway, in the common case, creating a `Realm` with the proper configuration is as simple as `let realm = try! Realm()`.
```swift
let package = realm.exportThreadSafePackage(containing: [foo, bar, baz])
queue.async {
    let realm = try! Realm()
    let items = realm.importOnCurrentThread(package)
}
```

2) If we _really_ want to rethink this API, we might structure it so that items can be exported and imported individually. We could probably optimize it so that it doesn't require pinning and unpinning the version for each item, but it might be tricky. We would however gain some type safety and it might be easier for users to think about the API this way. This implementation would not rely much on the current Object Store handover implementation (which was built for opaque handover packages), but it might still be worthwhile.
```swift
let fooRef = foo.threadSafeReference
let barRef = bar.threadSafeReference
queue.async {
    let realm = try! Realm()
    let foo = realm.threadSafeLookup(of: fooRef)
    let bar = realm.threadSafeLookup(of: barRef)
    // do stuff
}
```
and
```swift
let references = [foo, bar, baz].map { $0.threadSafeReference }
queue.async {
    let realm = try! Realm()
    let items = references.map(realm.threadSafeLookup)
    // do stuff
}
```